compiler/rustc_hir_typeck/src/upvar.rs

```diff
@@ -965,15 +969,15 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
 
         let mut obligations_should_hold = Vec::new();
         // Checks if a root variable implements any of the auto traits
-        for check_trait in auto_traits_def_id.iter() {
+        for (check_trait, rest) in auto_traits_def_id.iter() {
             obligations_should_hold.push(
                 check_trait
                     .map(|check_trait| {
                         self.infcx
                             .type_implements_trait(
                                 check_trait,
                                 ty,
-                                self.tcx.mk_substs_trait(ty, &[]),
+                                self.tcx.mk_substs_trait(ty, rest),
```

Comment:
why not `mk_substs_trait_with_effect`, rn we have an unused generic arg for `Clone`